Fix integer overflow during FEConvolveMatrix IPC decoder validator

ruthvikkonda opened the pull request:
#### 97d27fbddaca8412e0370afd58185195d60ec66e
<pre>
Fix integer overflow during FEConvolveMatrix IPC decoder validator
<a href="https://bugs.webkit.org/show_bug.cgi?id=300303">https://bugs.webkit.org/show_bug.cgi?id=300303</a>
<a href="https://rdar.apple.com/161647030">rdar://161647030</a>

Reviewed by Mike Wyrzykowski.

In the IPC decoder validator for FEConvolveMatrix, the kernelSize()-&gt;area()
overflows if given large width and height values. This causes a crash in GPUP.
The fix is to use unclampedArea() which will never overflow.

The fuzzer test case is altered slightly to consume the DidInitialize message
that gets sent back to WebContent to prevent it from reaching the dummy MessageReceiver
and hitting ASSERT_NOT_REACHED()

Test: ipc/decode-feConvolveMatrix-kernelSize-overflow.html
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow-expected.txt: Added.
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow.html: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/301246@main">https://commits.webkit.org/301246@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/edc4d33886ce242f71d6ebfe604d74f5ce84ef6f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125340 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45005 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35746 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132189 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77209 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b4581ec1-baf4-42ef-af94-dc7765f27cef) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45693 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53570 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95427 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63356 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/28aee5d2-52fd-4dfc-9518-15ce50a63f84) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128294 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36499 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112090 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75966 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35398 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30258 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75667 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106267 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30484 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134875 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52143 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39923 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103905 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52581 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108311 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103663 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26406 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49031 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27322 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49259 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52038 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57817 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51395 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54751 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53089 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/53476859a700824991f04458be1f4173b66f3451. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51927

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1e724d4b7033f0dca34411ffe61e1d61e40f4d61. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51927

ruthvikkonda commented:
The mac-AS-debug-wk2 bot will fail until my related PR https://github.com/WebKit/WebKit/pull/51979 lands. Once that lands, I can re-run that bot to rebase and it'll pass.

ruthvikkonda committed:
Fix integer overflow during FEConvolveMatrix IPC decoder validator
https://bugs.webkit.org/show_bug.cgi?id=300303
rdar://161647030

Reviewed by Mike Wyrzykowski.

In the IPC decoder validator for FEConvolveMatrix, the kernelSize()->area()
overflows if given large width and height values. This causes a crash in GPUP.
The fix is to use unclampedArea() which will never overflow.

The fuzzer test case is altered slightly to consume the DidInitialize message
that gets sent back to WebContent to prevent it from reaching the dummy MessageReceiver
and hitting ASSERT_NOT_REACHED()

Test: ipc/decode-feConvolveMatrix-kernelSize-overflow.html
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow-expected.txt: Added.
* LayoutTests/ipc/decode-feConvolveMatrix-kernelSize-overflow.html: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/301246@main
